app/encoding.go

```diff
@@ -1,14 +1,15 @@
 package app
 
 import (
-	"github.com/CosmWasm/wasmd/x/wasm"
-	"github.com/cosmos/cosmos-sdk/simapp"
 	simparams "github.com/cosmos/cosmos-sdk/simapp/params"
+	"github.com/cosmos/cosmos-sdk/std"
 )
 
 func MakeEncodingConfig() simparams.EncodingConfig {
-	encodingConfig := simapp.MakeEncodingConfig() // todo: this is the simapp !!!
-	wasm.RegisterCodec(encodingConfig.Amino)
-	wasm.RegisterInterfaces(encodingConfig.InterfaceRegistry)
+	encodingConfig := simparams.MakeEncodingConfig() // todo: this is the simapp !!!
```

Comment:
Should we not copy the type and the `simparams.MakeEncodingConfig()` ?

Also, this no longer registered all modules in the simapp (so ModuleBasics won't cause conflicts)? If tests pass, then this looks good, but seems a bit odd... I'm almost prefer just copying the simappparams code here so the whole flow is clear.

Reply:
agree, I will copy the code